src/compiler/types.ts

```diff
@@ -4664,6 +4664,7 @@ namespace ts {
         UseAliasDefinedOutsideCurrentScope      = 1 << 14,  // Allow non-visible aliases
         UseSingleQuotesForStringLiteralType     = 1 << 28,  // Use single quotes for string literal type
         NoTypeReduction                         = 1 << 29,  // Don't call getReducedType
+        OmitThisParameter                       = 1 << 30,

```

Comment:
For consistency, I think this flag should also be added to `TypeFormatFlags`, and then maybe look at `signatureToString`, `typeToString` etc to see if they need updating to be able to handle the new flag by omitting `this` params.

Reply:
@gabritto Thanks for the review. I've added `OmitThisParameter` to `TypeFormatFlags`